RadarSDK/RadarJSONCoding/RadarBeacon+CLBeacon.m

```diff
@@ -0,0 +1,43 @@
+
+#import "RadarBeacon+CLBeacon.h"
+#import "RadarLogger.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation RadarBeacon (CLBeacon)
+
+- (CLBeaconRegion *_Nullable)toCLBeaconRegion {
+    NSUUID *uuid = [self _NSUUID];
+    if (!uuid) {
+        NSString *message = [NSString stringWithFormat:@"Invalid beacon uuid %@", self.uuid];
+        [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:message];
+        return nil;
+    }
+    NSNumberFormatter *formatter = [[NSNumberFormatter alloc] init];
+    formatter.numberStyle = NSNumberFormatterDecimalStyle;
+    NSNumber *majorNumber = [formatter numberFromString:self.major];
+    NSNumber *minorNumber = [formatter numberFromString:self.minor];
+    return [[CLBeaconRegion alloc] initWithProximityUUID:uuid major:[majorNumber unsignedShortValue] minor:[minorNumber unsignedShortValue] identifier:self._id];
+}
+
+- (NSUUID *_Nullable)_NSUUID {
+    if (![self.uuid containsString:@"-"] && self.uuid.length == 32) {
+        @try {
+            NSString *firstComponent = [self.uuid substringWithRange:NSMakeRange(0, 8)];
+            NSString *secondComponent = [self.uuid substringWithRange:NSMakeRange(8, 4)];
+            NSString *thirdComponent = [self.uuid substringWithRange:NSMakeRange(12, 4)];
+            NSString *fourthComponent = [self.uuid substringWithRange:NSMakeRange(16, 4)];
+            NSString *lastComponent = [self.uuid substringWithRange:NSMakeRange(20, 12)];
+            NSString *formatedUUIDString = [NSString stringWithFormat:@"%@-%@-%@-%@-%@", firstComponent, secondComponent, thirdComponent, fourthComponent, lastComponent];
+            return [[NSUUID alloc] initWithUUIDString:formatedUUIDString];
+        } @catch (NSException *exception) {
+            return nil;
+        }
+    } else {
```

Comment:
do we want to do other validation here? like `length == 36`? There are a few strings that could fall through here that would be invalid

Reply:
we do the other validation in the server. https://github.com/radarlabs/server/blob/master/server/utils/AppleBeaconUtils.ts#L72

Basically after server validation, the uuid string must be either 1) a 32 bit UUID without hyphens or 2) a 36 bit UUID with hyphens. The logic here is just to converting scenario 1 to 2 (since this is a system requirement from Core location API). We also wrapped this inside a try-catch for extra safety